cvmfs/manifest_fetch.cc

```diff
@@ -162,4 +165,31 @@ Failures Fetch(const std::string &base_url, const std::string &repository_name,
   return result;
 }
 
+/**
+ * If the whitelist or the manifest are corrupted, fail-over once to another
+ * stratum 1 if more than a single stratum 1 is known.
+ */
+Failures Fetch(const std::string &base_url, const std::string &repository_name,
+               const uint64_t minimum_timestamp, const shash::Any *base_catalog,
+               signature::SignatureManager *signature_manager,
+               download::DownloadManager *download_manager,
+               ManifestEnsemble *ensemble)
+{
+  Failures result =
+    DoFetch(base_url, repository_name, minimum_timestamp, base_catalog,
+            signature_manager, download_manager, ensemble);
+  if ((result != kFailOk) && (result != kFailLoad) &&
+      (download_manager->num_hosts() > 1))
+  {
+    LogCvmfs(kLogCache, kLogDebug | kLogSyslogWarn,
+             "failed to fetch manifest (%d - %s), trying another stratum 1",
+             result, Code2Ascii(result));
+    download_manager->SwitchHost();
+    result =
+      DoFetch(base_url, repository_name, minimum_timestamp, base_catalog,
+              signature_manager, download_manager, ensemble);
+  }
+  return result;
```

Comment:
So this only tries one more Stratum 1?  Is it too difficult to try them all like happens with file downloads?